test/unit/org/apache/cassandra/db/compaction/CompactionControllerTest.java

```diff
@@ -236,6 +238,90 @@ public void testIgnoreOverlaps() throws Exception
         testOverlapIterator(false);
     }
 
+    static CountDownLatch memtableRaceStartLatch = new CountDownLatch(1);
+    static CountDownLatch memtableRaceFinishLatch = new CountDownLatch(1);
+
+    @Test
+    @BMRules(rules = {
+      @BMRule(name = "Pause between getting and processing partition",
+              targetClass = "org.apache.cassandra.db.partitions.TrieBackedPartition",
+              targetMethod = "create",
+              targetLocation = "AT ENTRY",
+              condition = "Thread.currentThread().getName().matches(\"[Cc]ompaction.*1\")",
+              action = "System.out.println(\"Byteman rule firing\");" +
+                       "org.apache.cassandra.db.compaction.CompactionControllerTest.memtableRaceStartLatch.countDown();" +
+                       "com.google.common.util.concurrent.Uninterruptibles.awaitUninterruptibly(" +
+                       "    org.apache.cassandra.db.compaction.CompactionControllerTest.memtableRaceFinishLatch, " +
+                       "    5, java.util.concurrent.TimeUnit.SECONDS);")
+    })
+    public void testMemtableRace() throws Exception
```

Comment:
For Review:

`CompactionControllerTest.testMemtableRace` fails.

Changing `MemtableParams` default memtable factory from `SkipListMemtableFactory` to `DefaultMemtableFactory` (TrieMemtable) will pass `testMemtableRace` **IF RUN BY ITSELF**. If the entire `CompactionControllerTest` is run in IntelliJ then it still fails. Not sure what's happening.

`MemtableThreadedTest` fails (hangs) on command line when executed with verify script** but passes in Intellij. Not sure what's happening there either.

** The verify script is just a script that finds tests modified by the commit being ported and executes them on the command line. The test names are put into a file and they are executed using:
```
ant testclasslist -Dtest.classlistprefix=unit -Dtest.classlistfile=testlist-unit.txt
```

Reply:
I opened https://github.com/riptano/cndb/issues/12218 for this test failure